### ISSUE TITLE
Update thread linking to external pages

### ISSUE THREAD

**Issue by timolegros** [hicommonwealth/commonwealth#4829]
## Description

In order to create a thread that links to an external website, the user needs to select 'Link' instead of 'Discussion' when creating a thread. This allows you to enter an arbitrary URL that will be added at the top of the thread page. This link cannot be updated or deleted.

Instead, adding external links should be done via a card on the right side of the page. Clicking a button in this card should open a model that allows a user to add or remove links (possibly including titles for each link).

## Engineering Requirements

- Remove option to create a "Link" thread instead of a "Discussion" thread.
- Add a new card on the thread page to allow adding or removing links.
- Use the `links` column rather than the `url` column in the `Threads` table to store external URL links per [Jake's comment](https://github.com/hicommonwealth/commonwealth/issues/4699#issuecomment-1682698715)

**Comment by jnaviask** [comment]
cc @zakhap to review

**Comment by CowMuon** [comment]
Blocker on #4699

**Comment by zakhap** [comment]
I would like to add a checkbox/selector to the modal that lets the user select the type of link they're providing. 

Story:
- As a user attaching a link to a proposal, I want to be able to designate that the link is a proposal, such that the tag that appears on the Thread Preview Row says "Proposal". I want this so I can visually distinguish between different types of linked content that make up the enriched thread experience.

**Comment by dillchen** [comment]
This ticket is good as scoped. And what I'm saying should not increase scope, nor block the ticket.

I just want to make sure that as this ticket is implemented that `createThread` (or w/e route is called now, allows all metadata objects to be associated on creation).

I will note, from 2.0 perspective as a prereq for #4447 to work well. We want to be able to support the creation or adding of _any_ internal link associated object (I.e Poll, Snapshot, or w/e) from the creation page.

The client side work will be handled from the "2.0 side" (e.g additional cards to add to the create page.)

**Comment by ianrowan** [comment]
Styling not complete, but thoughts on this UX: https://www.loom.com/share/1142d2342360422a83aa3b05afef36a3?sid=a258a7bd-dff9-49cf-b4f1-511a96950f05

No design to go off of here so this is what I thought up, feedback appreciated before PR 

CC @jnaviask  @MuonShot @timolegros

**Comment by ianrowan** [comment]
@jnaviask Ok, just saw Zaks comment when posting the prior comment. In that case do we want to have the general link UX plus a separate proposal link UX or blend the 2 with an option to select 'link types'? 

Both will require extension of the links model or potentially updating how we create identifiers for web proposals plus UI tags to identify them.

In theory can't the title field as built here be used to just create a title like 'Proposal #123' with the web links? 

<img width="324" alt="image" src="https://github.com/hicommonwealth/commonwealth/assets/36428666/5ffab7cf-6105-4391-9d16-6f16de255a71">

**Comment by jnaviask** [comment]
Curious if @dillchen or @sachben91 has thoughts here on product choices that lead us in the right direction for the future.

**Comment by CowMuon** [comment]
Removed `blocker` label; added `blocked` label @ianrowan

**Comment by CowMuon** [comment]
@jessmart1213 Do you have an answer for Ian on this to get him unblocked?

**Comment by ianrowan** [comment]
Thanks @CowMuon, in the meantime I will have the PR with the demoed functionality ready in case we decide we want to go with that to unblock Tim for now

**Comment by jessmart1213** [comment]
Speaking for design, I would prefer that all links, general and distinct links (proposals, linked threads, etc), have the same UX treatment at this moment. 

If we had more design bandwidth, I would say differentiating between the two links types would be ideal as to distinguish between general hyperlinks and the special functionality we have on Common that allows users to link in-product items to each other. But for now, let's keep it simple, with one UX treatment, as this is a nice-to-have and  not blocking user functionality. 

I will take Ian's comments into consideration though, for when its time to improve our linking experience.

**Comment by CowMuon** [comment]
If this unblocks you @ianrowan please remove the blocked tag; (don't want to presume here.)

**Comment by sachben91** [comment]
good with @jessmart1213 suggestion here
